analyzers/src/SonarAnalyzer.Common/Rules/BypassingAccessibilityBase.cs

```diff
@@ -26,16 +28,22 @@ public abstract class BypassingAccessibilityBase<TSyntaxKind> : SonarDiagnosticA
         where TSyntaxKind : struct
     {
         protected const string DiagnosticId = "S3011";
-        protected const string MessageFormat = "Make sure that this accessibility bypass is safe here.";
+        private const string MessageFormat = "Make sure that this accessibility bypass is safe here.";
 
-        protected FieldAccessTracker<TSyntaxKind> FieldAccessTracker { get; set;  }
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
-            FieldAccessTracker.Track(context,
+        protected FieldAccessTracker<TSyntaxKind> FieldAccessTracker { get; set; }
+
+        protected DiagnosticDescriptor Rule { get; }
+
+        protected BypassingAccessibilityBase(System.Resources.ResourceManager rspecStrings) =>
+            Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, rspecStrings);
+
+        protected override void Initialize(SonarAnalysisContext context) =>
+            FieldAccessTracker.Track(
+                context,
```

Comment:
I like a lot these refactorings but I think we should do them in separate PRs if they are not related to the current issue. Having a single topic for a PR makes both the review and troubleshooting (checking the history for the reason the code has changed) easier.